### DISCUSSION TITLE
Suggestions on setting up environment for contribution?

### DISCUSSION THREAD

**Discussion by friday963** [pouriyajamshidi/tcping#147] [General]
Hello,  I'm interested in contributing but not having success pulling down the project and running it locally.  Is there any suggested steps I can take to set up my environment so I can pull down the project, set a breakpoint and step through the project?  `go run tcping.go` is failing and I'm not sure what steps to take to use the program in this fashion.

**Comment by pouriyajamshidi** [comment]
Hi @friday963,

Thanks for considering to contribute and reaching out.

I am not entirely sure why that could be, since we are having many different people already contributing without seeing errors. This repo does not have any specific security setup. 

Perhaps you are trying to clone it using SSH? In that case it will not work out.

My suggestion would be to **fork** this repo, then clone your fork locally and then create a branch in which you can break and fix stuff.

Whenever you are happy with your work or perhaps you have completed an issue, you can create a pull request to this repo.

Let me know if you get stuck somewhere.

If you could share the output of the error you face, that'd help.

Also you can run it like: `go run ./... www.example.com 443`

cheers

**Comment by friday963** [comment]
Hello, @pouriyajamshidi it wasn't so much about pulling the project down from github, but rather setting up the environment.  

Either way, I had originally tried doing it in WSL which is what prompted my question, but I pulled it down onto my VM host and its working.  No issues on my side at this point 👍

**Reply by pouriyajamshidi** [reply]
Hey @friday963,

Glad to hear that mate 👍🏻
